update react select to web component and tests

ataker opened the pull request:
## Summary

- Update react select component to web component
- Update tests

## Related issue(s)

- [Replace select component](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1735)

## Testing done

- Unit tests updated

## What areas of the site does it impact?

Only modal used in dev mode impacted

## Acceptance criteria

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

ataker committed:
update react select to web component and tests

jamigibbs commented on src/platform/forms/save-in-progress/SaveInProgressDevModal.jsx:
```diff
@@ -116,13 +118,13 @@ const SipsDevModal = props => {
               value={{ value: sipsUrl }}
               includeBlankOption={false}
-              onValueChange={value => setSipsUrl(value.value)}
+              onVaSelect={value => setSipsUrl(value.value)}
```
Does the returning value actually work? The structure in a web component is `event.detail.value` for a custom event https://design.va.gov/about/developers/using-web-components#custom-events

ataker committed:
Select replace test (#24486)

* [eFSR] #59517 Accessibility testing color and contrast bug (#24409)

* Color contrast ratio fix for mini summary card delete button hover.

* Profile/aw/55398/remove duplicate ga event mil info (#24412)

* fix: remove duplicate GA event

* chore: clean up import CONTACTS

* chore: cleanup markup to remove brs

* fix: update import for CONTACTS

* MHV-45952 Mhv medications manual set up (#24399)

* added template files for medications start app; app entry file, reducer, route, sass, and landing component called app.

* added codeownders file.

* fixed routing issue, changed sass file to .scss instead of .sass, added TODO for import fixes

* My VA - Remove deprecated v0/appointments code (#24297)

* VACMS-13590 GMT < NMT Jonesboro case for Income Limits (#24356)

* 57110 Update phone number on alert (#24198)

* 57110 Update phone number on alert

* 57110 use tty param

* 56920 Final, approved content

* 56920 fix case

* Fixes VAOS PastAppointmentsListV2 unit test time zone issue (#24413)

* Moves dates inside of assertions

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

* Converts test dates to function

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

---------

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

* VACMS-13738 Add year screen for Income Limits (#24411)

* Supplemental Claims | Fix confirmation page focus (#24414)

* Fix confirmation & some page focus

* Update unit tests

* Create conditional for Child Educational Expenses page in HCA Household V2 (#24421)

* Dhp 878 update platform to pathway (#24422)

* DHP-878 update dhp container test to use 'pathway'

Co-authored-by: Yoni Doron-Peters <jonathan.d@thoughtworks.com>

* DHP-878 update dhp page to use 'pathway'

Co-authored-by: Yoni Doron-Peters <jonathan.d@thoughtworks.com>

---------

Co-authored-by: Yoni Doron-Peters <jonathan.d@thoughtworks.com>

* Remove organic adoption experiment modal (#24423)

* [#MHV-44893] Radiology Single Image Page (#24420)

* [#MHV-44893] Radiology Single Image Page

* [#MHV-44893] Added Unit test

* [#MHV-44893] Added Aria hidden tags

* Updated title string on details page to use hyphen key. (#24403)

* Updated title string on details page to use hyphen key.

* updated heading string in details test class.

* Prevent questions from being viewed after checkin (#24373)

* Prevent questions from being viewed after checkin

* Fix unit test

* Remove unnecessary test and describe block

* Add cypress test for back button

* Pass router down to check for check in complete once

* Test prevent navigation to questions

* Pass in true if app is pre-check-in

* Pass router from pre-check-in pages

* Fix import

* Fix linting errors

* [eFSR] Remove unnecessary fieldsets #59289 (#24408)

* Removed unnecessary fieldset and changed legend to h3

* Made h3 conditional to follow the heading structure on the form page vs. the review page.

* Adding correct spacing to titles.

* Supplemental Claim | Fix confirmation page initial focus for Safari (#24424)

* Add delay to Confirmation page initial focus for Safari

* Use waitForRenderThenFocus function

* [CST] Update AskVAPage to use Lighthouse version of submitRequest function (#24385)

* Updating AskVAPage to use Lighthouse version of submitRequest function

* [eFSR] #59524 Content zoom and reflow fix for dependents input page  (#24415)

* Fixed content overflow issue on 300% and up for dependents input question.

* Using widgetClassNames instead to target input with and removing extra custom class.

* Fixed input widths to have label not wrap and look really squished.

* Added fieldset/legend to Installment Contract page.

* Added currency label to compromise resolution input

* Added more padding to the left of the label to match the spacing DS radio button input/label and changed the font-size of the label to base.

* Removed unnecessary verbiage.

* 21-4142 confirmation page fixes (#24425)

* Remove widths on need help box

* Fix focus state for confirmation page

* [eFSR] Mini Summary Navigation Fixes (#24405)

* Simple forms - hide 21-4142 when flipper off (#24391)

* add flag to hide form if disabled

* content updates

* plural forms

* content updates

* update url per IA review (#24430)

* modify CI workflow so that dependabot PRs don't fail (#24426)

* modify workflow so that dependabot PRs don't fail

* fix typo

* 59011 Remove fetchPatientEligibility function for V0 (#24345)

* remove fetchPatientEligibilityFromVAR function
#53093
#53095

* fix unit test

* Add custom tracking names to GA4 (#24404)

* Update custom tracking name

* unit test

* Simple forms: Make WIP component shared and move content to constants (#24433)

* make WIP component shared and move content to constants

* default to form in local

* 53085 Remove getPendingAppointment function for v0 (#24334)

* remove v0 getPendingAppointment

* update unit test

* update unit test

* update unit test

* update unit test

* add new e2e mock function

* fix e2e test

* fix e2e test

* fix e2e test

* fix e2e test

* fix e2e test

* [CST] Supporting documents and Tracked items are now sorted by date (#24427)

* Sorting tracked items and supporting documents by date

* Removing empty p when there is no description

* Updating unit test for serializer

* VA-13677: Remove COVID Status from VAMC Facilities (#24364)

* VA-13673: Remove COVID Status from VAMC Facilities

* VA-13677: Remove VA Locator intro text

* VA-13673: Remove Unused LocationCovidStatus component

* refactor confirmation page view to be reusable (#24438)

* Notice of Disagreement | Use platform contact definition (#24389)

* NOD: Use platform profileContactInfo

* Add contact info UI validation

* Update unit tests

* Fix e2e tests

* Fix profile imports

* Fix va-telephone prop

* Revert va-telephone change

* Fix imports

* More import fixes

* Remove unused review description

* Update unit tests

* Remove unit test

* 58307 update documents needed logic (#24442)

* Sorting tracked items and supporting documents by date

* Removing empty p when there is no description

* Removing unwanted characters from tracked item descriptions

* Mhv 41588 error on compose (#24387)

* [MHV-41314] fixed issue where the delete option would bring up 2 pop ups.

* [MHV-41314] cleared navigation errors on autosave to allow user to navigate away after autosave, and not trigger the RouteLeaveGuard prompt.

* [MHV 41588] added logic to prompt the user if there is an attachment and the user tries to leave.

* [MHV 41588] removed body text if the navigation error is being triggred by the user adding an attachment.

* modify workflows to work with dependabot PRs (#24439)

* modify workflows to work with dependabot PRs

* fix typo0

* alter config to support multiple hosts (#24447)

Co-authored-by: Curt Bonade <cbonade@Curts-MacBook-Pro.local>

* [eFSR] Fix - Hiding mobile number extension field (#24435)

* VACMS-13955 Homepage post-launch cleanup (#24444)

* Mhv 44996 prescription details (#24448)

* [#MHV-44996] Fixed router issue

* [#mhv-44996] Full details page

* [#mhv-44996] Unit test and data-test-id

* [#mhv-44996] style changes

* Add review errors (#24451)

* MHV-42057: Microbio link to api work done (#24450)

* changed letters generator request type for download endpoint to POST (#24376)

* My VA - Payment info cleanup (#24396)

* removed showPaymentAndDebtSection feature toggle

* committing a successful test before i work on the
rest

* updated e2e tests that were using the removed
showPaymentAndDebt feature toggle

* remove e2e tests for old section

* - using .findAllByTestId instad of .findByTestId
or .get
- updating some test names to be more specific
- removed "feature is hidden" tests

* - added back payment history state check for
BenefitPaymentsV2

* fix eslint issue

* Update UI for DSLogon and MHV users (#24386)

* feat: cond display email in contact and tests

* chore: change email to information in link

* fix: eslint warning

* fix: eslint warnings

* Update HCA Household V2 dependent list card from horizontal to vertical stack view (#24454)

* 54747 ds web component forms library integration (#24456)

* Merge RJSF v3 web component integration branch 

* Allow for specifying 'ui:webComponentField' in RJSF e.g. VaRadioField, VaTextInputField, etc..

* Added new folder platform/forms-system/src/js/web-component-fields
  - SsnField, VaCheckboxField, VaMemorableDateField, VaRadioField, VaTextInputField, YesNoField

* Added new folder platform/forms-system/src/js/web-component-patterns
  - addressUI/schema, dateUI/schema, emailUI/schema, fullNameUI/schema, etc

---------

Co-authored-by: Nick Sprinkle <ndsprinkle@gmail.com>

* MHV-42057: Lab and test details fix (#24455)

* Update max length for address fields in 686c. (#24453)

We've been seeing errors for the address_line1 and city fields when submitting data to RBPS/BIS indicating that the max length for these fields should be 35 and 30, respectively. This PR updates the max length accordingly.

* 56920 First pass of sync 526EZ online form with paper form (#24315)

* 56920 First pass of 526 sync

* 56920 Ensure ancillary views don't display unnecessarily and move 8940

* 56920 cleanup warnings

* 56920 update configs for moved pages

* 56920 cleanup imports

* 56920 Adding one more check to prevent the additional disability benefits summary from displaying if we're not using the ancillary wizard

* 56920 reverting changes for ancillary form and individual unemployability

* Bump loader-utils from 1.4.0 to 1.4.2 (#22700)

Bumps [loader-utils](https://github.com/webpack/loader-utils) from 1.4.0 to 1.4.2.
- [Release notes](https://github.com/webpack/loader-utils/releases)
- [Changelog](https://github.com/webpack/loader-utils/blob/v1.4.2/CHANGELOG.md)
- [Commits](https://github.com/webpack/loader-utils/compare/v1.4.0...v1.4.2)

---
updated-dependencies:
- dependency-name: loader-utils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ian <harrison.ian.t@gmail.com>

* Bump ua-parser-js from 0.7.32 to 0.7.35 (#23884)

Bumps [ua-parser-js](https://github.com/faisalman/ua-parser-js) from 0.7.32 to 0.7.35.
- [Release notes](https://github.com/faisalman/ua-parser-js/releases)
- [Changelog](https://github.com/faisalman/ua-parser-js/blob/master/changelog.md)
- [Commits](https://github.com/faisalman/ua-parser-js/compare/0.7.32...0.7.35)

---
updated-dependencies:
- dependency-name: ua-parser-js
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ian <harrison.ian.t@gmail.com>

* Bump webpack from 5.70.0 to 5.76.0 (#23604)

Bumps [webpack](https://github.com/webpack/webpack) from 5.70.0 to 5.76.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.70.0...v5.76.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ian <harrison.ian.t@gmail.com>

* Bump @actions/core from 1.8.0 to 1.9.1 (#21975)

Bumps [@actions/core](https://github.com/actions/toolkit/tree/HEAD/packages/core) from 1.8.0 to 1.9.1.
- [Release notes](https://github.com/actions/toolkit/releases)
- [Changelog](https://github.com/actions/toolkit/blob/main/packages/core/RELEASES.md)
- [Commits](https://github.com/actions/toolkit/commits/HEAD/packages/core)

---
updated-dependencies:
- dependency-name: "@actions/core"
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ian <harrison.ian.t@gmail.com>

* Bump css-what from 2.1.0 to 2.1.3 (#22346)

Bumps [css-what](https://github.com/fb55/css-what) from 2.1.0 to 2.1.3.
- [Release notes](https://github.com/fb55/css-what/releases)
- [Commits](https://github.com/fb55/css-what/compare/v2.1.0...v2.1.3)

---
updated-dependencies:
- dependency-name: css-what
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ian <harrison.ian.t@gmail.com>

* Bump terser from 5.12.1 to 5.14.2 (#21722)

Bumps [terser](https://github.com/terser/terser) from 5.12.1 to 5.14.2.
- [Release notes](https://github.com/terser/terser/releases)
- [Changelog](https://github.com/terser/terser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terser/terser/commits)

---
updated-dependencies:
- dependency-name: terser
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ian <harrison.ian.t@gmail.com>

* Bump @xmldom/xmldom from 0.7.5 to 0.7.11 (#24460)

Bumps [@xmldom/xmldom](https://github.com/xmldom/xmldom) from 0.7.5 to 0.7.11.
- [Release notes](https://github.com/xmldom/xmldom/releases)
- [Changelog](https://github.com/xmldom/xmldom/blob/0.7.11/CHANGELOG.md)
- [Commits](https://github.com/xmldom/xmldom/compare/0.7.5...0.7.11)

---
updated-dependencies:
- dependency-name: "@xmldom/xmldom"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ian <harrison.ian.t@gmail.com>

* feat: add analytics for lighthouse direct deposit (#24452)

* Sa 61735 61736 form flow fix (#24466)

* intermediate commit adds ui:required logic for validation fields

* intermediate commit adds additional ui:options for name related formFields on ui:schema

* Adds custom Review Page component for Identity view

* Adds custom Review Page component for Identity view

* Adds new component props to submit-transform with fallback, cleans up formConfig and review Page

* Updates TOE form.js to include both view component and form fields

* Adds Custom Review Page to account for new component with fallback to formFields

* Adjusts properties being accessed in submissionForm in MEB and in data within TOE review page

* Adjusts constants for nesting in submissionForm

* Adds fallback for userFullName on confirmation page prop

* Adds loaded data in submission when view component is showing

* Removes spec that for applicant information page that uses html elements that no longer exist with new component

* intermediate commit adds ui:required logic for validation fields

* intermediate commit adds additional ui:options for name related formFields on ui:schema

* Adds custom Review Page component for Identity view

* Adds custom Review Page component for Identity view

* Adds new component props to submit-transform with fallback, cleans up formConfig and review Page

* Updates TOE form.js to include both view component and form fields

* Adds Custom Review Page to account for new component with fallback to formFields

* Adjusts properties being accessed in submissionForm in MEB and in data within TOE review page

* Adjusts constants for nesting in submissionForm

* Adds fallback for userFullName on confirmation page prop

* Adds loaded data in submission when view component is showing

* Removes spec that for applicant information page that uses html elements that no longer exist with new component

* Removes cypress test for your informaiton review page panel because the component is now customized

* Revert "modify workflows to work with dependabot PRs (#24439)" (#24461)

This reverts commit 203c9d10227cfb933774be98972795571afdf0ab.

* Revert "modify CI workflow so that dependabot PRs don't fail (#24426)" (#24462)

This reverts commit 8f5ba2eaa18ec34a555ef6100bb61294419c49bd.

* Updating sorting algorithm (#24457)

* Update event response value

* Add select options

* VACMS-13588 Fixing breadcrumbs to use va-breadcrumbs (#24463)

* Add condition check for options

---------

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>
Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: Joseph Lee <joseph@cityfriends.tech>
Co-authored-by: Adam Whitlock <adam.whitlock@adhocteam.us>
Co-authored-by: robertbylight <107279507+robertbylight@users.noreply.github.com>
Co-authored-by: Derrick Ellerbie <dellerbie@users.noreply.github.com>
Co-authored-by: Randi Mays <randimays1024@gmail.com>
Co-authored-by: Christine <127446042+christinec-fftc@users.noreply.github.com>
Co-authored-by: Ryan Shaw <ryan.shaw@adhocteam.us>
Co-authored-by: Rob Garrison <wowmotty@gmail.com>
Co-authored-by: Matt Long <longmd@me.com>
Co-authored-by: Joseph Presley <joseph.presley@thoughtworks.com>
Co-authored-by: Yoni Doron-Peters <jonathan.d@thoughtworks.com>
Co-authored-by: Ksenia Coulter <kseniacoulter@gmail.com>
Co-authored-by: Kolby Balch <108146636+KolbyVA@users.noreply.github.com>
Co-authored-by: Brian Seek <brian.seek@civicactions.com>
Co-authored-by: John Woolschlager <wullaski@gmail.com>
Co-authored-by: Jerek Shoemaker <jerek.shoemaker@adhocteam.us>
Co-authored-by: Jeana <117098918+jeana-adhoc@users.noreply.github.com>
Co-authored-by: Andrew Rodiek <andrew.rodiek@gmail.com>
Co-authored-by: Garrett <57802560+gmrabian@users.noreply.github.com>
Co-authored-by: Ian <ian.harrison@adhocteam.us>
Co-authored-by: Tony Wiliiams <tony.williams@va.gov>
Co-authored-by: jenniemc <54327023+jenniemc@users.noreply.github.com>
Co-authored-by: Max Antonucci <maxx1128@users.noreply.github.com>
Co-authored-by: cbonade <34250637+CBonade@users.noreply.github.com>
Co-authored-by: Curt Bonade <cbonade@Curts-MacBook-Pro.local>
Co-authored-by: mattwrightva <107576133+mattwrightva@users.noreply.github.com>
Co-authored-by: samcoforma <105806934+samcoforma@users.noreply.github.com>
Co-authored-by: allisonlu <allisonlu820@gmail.com>
Co-authored-by: Robert Hasselle <123402053+rhasselle-oddball@users.noreply.github.com>
Co-authored-by: Nick Sprinkle <ndsprinkle@gmail.com>
Co-authored-by: Eugene Lynch <eugenestapler@gmail.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ian <harrison.ian.t@gmail.com>
Co-authored-by: Brad Bergeron <bradley.bergeron@accenturefederal.com>
Co-authored-by: Jami Gibbs <jami.gibbs@adhocteam.us>

ataker committed:
Merge branch 'main' into replace-react-selct-component-dst1735

ataker committed:
Merge branch 'main' into replace-react-selct-component-dst1735
